openactive/openactive-test-suite
# feat(broker): /is-order-uuid-present API

Which returns whether or not a given Order UUID exists in the specified feed

This is for the `booking-partner-authentication` tests from https://github.com/openactive/openactive-test-suite/issues/441

## QA

I've tested the following scenarios:

1. Turn on broker, open `http://localhost:3000/is-order-uuid-present/orders/primary/abc` before feeds have started harvesting

    This requests waits until broker finishes harvesting feeds and then returns `false` ✅ 
2. Turn on broker, wait for feeds to finish harvesting, open `http://localhost:3000/is-order-uuid-present/orders/primary/abc`

    Immediately returns `false` ✅ 
3. Turn on broker, run the `seller-requested-cancellation` test, which should place an Order in the feed, check the UUID of the generated Order using the test results. The ID was `da32a390-267b-42fd-9748-27600b29837e`. Then, open `http://localhost:3000/is-order-uuid-present/orders/primary/da32a390-267b-42fd-9748-27600b29837e`

    Immediately returns `true` ✅ 
4. After # 3, restart Broker and open `http://localhost:3000/is-order-uuid-present/orders/primary/da32a390-267b-42fd-9748-27600b29837e` before it finishes harvesting feeds.

    Request waits until Broker has finished harvesting feeds and then returns `true` ✅ 
5. After # 3, try requests for different UUIDs and for the `secondary` booking partner.

    Confirm that these all return `false`. ✅ 
6. After # 3, also check that it works for the `order-proposals` feed with one of the approval runs of the same test.

    Again, worked as expected - returned `true` ✅
